src/js/elements/data_mapper/components/indicator_tree_view.js

```diff
@@ -0,0 +1,203 @@
+import React, {useState, useCallback, useMemo, useEffect} from "react";
+
+import {
+  StyledCategoryTreeItem,
+  StyledSubCategoryTreeItem,
+  StyledSubindicatorTreeItem,
+  StyledIndicatorTreeItem
+} from "./styledElements";
+import Box from "@mui/material/Box";
+import {Typography} from "@mui/material";
+import VisibilityIcon from '@mui/icons-material/Visibility';
+import VisibilityOffIcon from '@mui/icons-material/VisibilityOff';
+
+
+const LoadingItemView = (props) => {
+  return (
+    <StyledSubindicatorTreeItem nodeId={"loading"} label={
+      <Box sx={{ display: 'flex', alignItems: 'center', p: 0.5, pr: 0 }}>
+        <Typography variant="body2" sx={{ fontSize: '1em', letterSpacing: '.3px', color: '#666'}}>
+          Loading...
+        </Typography>
+      </Box>
+    }/>
+  )
+}
+
+const SubindicatorItemView = (props) => {
+
+  const onClickSubindicator = useCallback(
+    () => {
+      props.controller.onSubIndicatorClick({
+          indicatorTitle: props.indicator.label,
+          selectedSubindicator: props.subindicator,
+          parents: props.parents,
+          choropleth_method: props.indicator.choropleth_method,
+          indicatorId: props.indicator.id,
+          indicatorData: props.indicatorData,
+          versionData: props.indicator.version_data,
+          metadata: {
+            ...props.indicator.metadata,
+            indicatorDescription: props.indicator.description,
+          },
+          config: {
+              choroplethRange: props.indicator.choropleth_range,
+              enableLinearScrubber: props.indicator.enable_linear_scrubber,
+              chartConfiguration: props.indicator.chartConfiguration
+          }
+      })
+    }, [
+      props.controller,
+      props.indicator,
+      props.subindicator
+    ]
+  )
+
+  return (
+    <StyledSubindicatorTreeItem nodeId={props.subindicator} label={
+      <Box sx={{ display: 'flex', alignItems: 'center', p: 0.5, pr: 0 }}>
+        <Typography variant="body2" sx={{ fontSize: '1em', letterSpacing: '.3px', color: '#666'}} className={"truncate"}>
+          {props.subindicator}
+        </Typography>
+      </Box>
+    } onClick={(e) => onClickSubindicator()} data-test-id={props.subindicator} className={"subIndicator-item"}/>
+  )
+}
+
+const IndicatorItemView = (props) => {
```

Comment:
is there a reason we don't have separate classes for category, subCategory, indicator and subIndicator?

Reply:
My thinking behind this was : subCategory, indicator and subIndicator components won't get used anywhere else and we didn't export them too.
So i put them in same file to avoid one more folder inside components.
But I have no issues separating it out if you think it will be cleaner.

Reply:
@emre2038

Reply:
I generally separate them in such cases but I'm ok with this too

Reply:
cool .. Will try to clean up in next ticket 👍